lib/modules/datasource/pypi/index.ts

```diff
@@ -43,45 +46,51 @@ export class PypiDatasource extends Datasource {
     packageName,
     registryUrl,
   }: GetReleasesConfig): Promise<ReleaseResult | null> {
-    let dependency: ReleaseResult | null = null;
     // TODO: null check (#22198)
-    const hostUrl = ensureTrailingSlash(
-      registryUrl!.replace('https://pypi.org/simple', 'https://pypi.org/pypi'),
-    );
+    const hostUrl = ensureTrailingSlash(registryUrl!);
+    const simpleHostUrl = hostUrl.replace(jsonPyPiUrl, simplePyPiUrl);
+    const pypiJsonHostUrl = hostUrl.replace(simplePyPiUrl, jsonPyPiUrl);
     const normalizedLookupName = normalizePythonDepName(packageName);
 
-    // not all simple indexes use this identifier, but most do
-    if (hostUrl.endsWith('/simple/') || hostUrl.endsWith('/+simple/')) {
+    const simpleDependencies = await this.getResultsViaSimple(
+      normalizedLookupName,
+      simpleHostUrl,
+    ).catch((err) => {
       logger.trace(
-        { packageName, hostUrl },
-        'Looking up pypi simple dependency',
+        { packageName, hostUrl: simpleHostUrl },
+        'Simple api not found. Looking up pypijson api as fallback.',
       );
-      dependency = await this.getSimpleDependency(
-        normalizedLookupName,
-        hostUrl,
+      return null;
```

Comment:
This appears to return early with `null` and not use pypijson as fallback

Reply:
I think the return here ends up the value of `simpleDependencies`.